### PR TITLE
TS-19-v2-25: Fix ESLint errors in share directory

### LINKED ISSUE #25: Refactor ESLint error in share directory
Acceptance Criteria:
Fix all ESLint errors in share directory

Estimate: 1 day | 2 days

### PR DESCRIPTION
**Pull request by HiroFive**

### CODE REVIEW

File: frontend/src/share/modals/modal/setDefaultDialog.js [hunk 1]
```diff
@@ -18,7 +18,11 @@ export const SetChangeDialog = (props) => {
     const handleClose = () => {
         onClose(cardId);
     };
-    const className = 'set-default';
+    const ifHideTitle =
+        isHide === disabledCard.HIDE
+            ? i18n.t('common:do_you_wanna_disable')
+            : i18n.t('common:do_you_wanna_show');
```
Comment (Akr1s) [inline frontend/src/share/modals/modal/setDefaultDialog.js:24]: Make a separate function for reusable logic

File: frontend/src/share/modals/modal/showStudentsDialog.js [hunk 2]
```diff
@@ -9,196 +9,156 @@ import Button from '@material-ui/core/Button';
...
     const setGroupsOption = () => {
-        return groups !== undefined
-            ? groups.map((item) => {
-                  return { value: item.id, label: `${item.title}`, ...item };
-              })
-            : null;
+        if (!isNil(groups)) {
+            return groups.map((item) => ({ value: item.id, label: `${item.title}`, ...item }));
+        }
+        return [];
```
Comment (Akr1s) [inline frontend/src/share/modals/modal/showStudentsDialog.js]: Maybe you should return null instead?

File: frontend/src/share/modals/modal/showStudentsDialog.js [hunk 3]
```diff
@@ -9,196 +9,156 @@ import Button from '@material-ui/core/Button';
...
     const handleCheckElement = (event) => {
-        const studentsTmp = [...checkBoxStudents];
-        studentsTmp.forEach((student) => {
-            if (student.id === Number(event.target.value)) student.checked = event.target.checked;
-        });
-        setCheckBoxStudents(studentsTmp);
+        setCheckBoxStudents(
+            checkBoxStudents.map((checkbox, index) => {
+                if (checkbox.id === Number(event.target.value)) {
+                    checkBoxStudents[index].checked = event.target.checked;
```
Comment (Akr1s) [inline frontend/src/share/modals/modal/showStudentsDialog.js:57]: isn't `checkBoxStudents[index]` the same as `checkbox`?

File: frontend/src/share/modals/modal/showStudentsDialog.js [hunk 4]
```diff
@@ -9,196 +9,156 @@ import Button from '@material-ui/core/Button';
...
-        const studentsTmp = [...checkBoxStudents];
-        studentsTmp.forEach((student) => {
-            if (student.checked) {
-                student.checked = false;
-            }
-        });
-        setCheckBoxStudents(studentsTmp);
+        setCheckBoxStudents(
+            checkBoxStudents.map((checkbox, index) => {
+                checkBoxStudents[index].checked = false;
```
Comment (Akr1s) [inline frontend/src/share/modals/modal/showStudentsDialog.js:85]: the same as above

File: frontend/src/share/modals/modal/showStudentsDialog.js [hunk 5]
```diff
@@ -9,196 +9,156 @@ import Button from '@material-ui/core/Button';
...
     const setSelectDisabled = () => {
         let resChecked = true;
-        for (let i = 0; i < checkBoxStudents.length; i++) {
-            if (isChecked(checkBoxStudents[i])) {
+        for (let i = 0; i < checkBoxStudents.length; i += 1) {
+            if (checkBoxStudents[i].checked) {
                 resChecked = false;
                 break;
             }
         }
```
Comment (Akr1s) [inline frontend/src/share/modals/modal/showStudentsDialog.js]: you could use `Array.prototype.every()` instead

**Review by Akr1s — CHANGES_REQUESTED** [review]

File: frontend/src/share/modals/modal/showStudentsDialog.js (hunk 2 above)
Comment (HiroFive) [reply frontend/src/share/modals/modal/showStudentsDialog.js]: no because we generate options for select

**Review by Niknikson — APPROVED** [review]

File: frontend/src/share/modals/modal/showStudentsDialog.js (hunk 3 above)
Comment (HiroFive) [reply frontend/src/share/modals/modal/showStudentsDialog.js:57]: `checkBoxStudents[index]` so we avoid no-param-reassign

File: frontend/src/share/modals/modal/showStudentsDialog.js (hunk 4 above)
Comment (HiroFive) [reply frontend/src/share/modals/modal/showStudentsDialog.js:85]: the same

File: frontend/src/share/modals/modal/showStudentsDialog.js (hunk 5 above)
Comment (HiroFive) [reply frontend/src/share/modals/modal/showStudentsDialog.js]: Then we won't be able to stop `every` through `break`

File: frontend/src/share/modals/modal/setDefaultDialog.js (hunk 1 above)
Comment (HiroFive) [reply frontend/src/share/modals/modal/setDefaultDialog.js:24]: this logic fixed in #36

**Review by Akr1s — APPROVED** [review]

**Review by ViktoriiaKosovan — APPROVED** [review]

**Pull request closed without merging**